x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DateMathException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic;
+
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xpack.esql.EsqlClientException;
+import org.elasticsearch.xpack.ql.tree.Source;
+
+/**
+ * Represents e.g. overflows when folding temporal values caused by a faulty query.
+ * This is turned into a REST response with HTTP status (400).
+ * Even though this is a pretty specific exception, we cannot use e.g. a anonymous class that inherits from {@link EsqlClientException},
+ * because then the error message sent back in the REST response would have an empty `type` string.
+ */
+public class DateMathException extends EsqlClientException {
```

Comment:
@costin : As user-facing date math exceptions need to be thrown in more than 1 class now, I added this. This seemed to be the most pragmatic solution, but we can also resort to an anonymous class inheriting from EsqlClientException (with empty `type` attribute in the REST response, though), or make `EsqlClientException` non-abstract.